e2e/bitrise.yml

```diff
@@ -3,169 +3,172 @@ default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
-    - BITRISE_APPLE_APPLE_CERTIFICATE_URL_LIST: $BITRISE_APPLE_APPLE_CERTIFICATE_URL_LIST
-    - BITRISE_APPLE_APPLE_CERTIFICATE_PASSPHRASE_LIST: $BITRISE_APPLE_APPLE_CERTIFICATE_PASSPHRASE_LIST
-    - IONIC_USERNAME: $IONIC_USERNAME
-    - IONIC_PASSWORD: $IONIC_PASSWORD
+  # Shared test configs
+  - TEST_APP_URL: https://github.com/bitrise-io/ionic-conference-app.git
+  - TEST_APP_BRANCH: master
+  # Shared test secrets
+  - BITRISE_APPSTORECONNECT_API_KEY_URL: $BITRISE_APPSTORECONNECT_API_KEY_URL
+  - BITRISE_APPSTORECONNECT_API_KEY_ISSUER_ID: $BITRISE_APPSTORECONNECT_API_KEY_ISSUER_ID
+  - BITRISE_APPLE_APPLE_CERTIFICATE_URL_LIST: $BITRISE_APPLE_APPLE_CERTIFICATE_URL_LIST
+  - BITRISE_APPLE_APPLE_CERTIFICATE_PASSPHRASE_LIST: $BITRISE_APPLE_APPLE_CERTIFICATE_PASSPHRASE_LIST
+  - IONIC_USERNAME: $IONIC_USERNAME
+  - IONIC_PASSWORD: $IONIC_PASSWORD
 
 workflows:
   test_ios_device_build:
     before_run:
     - _check_xcode_available
     steps:
     - bitrise-run:
-        run_if: '{{enveq "XCODE_AVAILABLE" "true"}}'
+        run_if: |-
+          {{ enveq "XCODE_AVAILABLE" "true" }}
         inputs:
         - workflow_id: utility_test_ios_device_build
         - bitrise_config_path: ./e2e/bitrise.yml
 
   utility_test_ios_device_build:
     envs:
-      - TARGET: device
-      - PLATFORM: ios
+    - TARGET: device
+    - PLATFORM: ios
     before_run:
-      - _setup
-      - _install_dependencies_with_npm
+    - _setup
+    - _install_dependencies_with_npm
     steps:
-      - ionic-prepare:
-          inputs:
-            - workdir: $BITRISE_SOURCE_DIR/_tmp
-            - platform: ios
-            - ionic_version: latest
-      - ios-auto-provision-appstoreconnect:
-          inputs:
-            - distribution_type: development
-            - project_path: $BITRISE_SOURCE_DIR/_tmp/platforms/ios/Ionic Conference App.xcworkspace
-            - scheme: Ionic Conference App
-            - connection: 'off'
-            - api_key_path: $BITRISE_APPSTORECONNECT_API_KEY_URL
-            - api_issuer: $BITRISE_APPSTORECONNECT_API_KEY_ISSUER_ID
-            - certificate_urls: $BITRISE_APPLE_APPLE_CERTIFICATE_URL_LIST
-            - passphrases: $BITRISE_APPLE_APPLE_CERTIFICATE_PASSPHRASE_LIST
-      - generate-cordova-build-configuration:
-          inputs:
-            - configuration: release
-            - development_team: $BITRISE_DEVELOPER_TEAM
-            - code_sign_identity: $BITRISE_DEVELOPMENT_CODESIGN_IDENTITY
-            - provisioning_profile: $BITRISE_DEVELOPMENT_PROFILE
-            - package_type: $BITRISE_EXPORT_METHOD
+    - ionic-prepare:
+        inputs:
+        - workdir: $BITRISE_SOURCE_DIR/_tmp
+        - platform: ios
+        - ionic_version: latest
+    - ios-auto-provision-appstoreconnect:
+        inputs:
+        - distribution_type: development
+        - project_path: $BITRISE_SOURCE_DIR/_tmp/platforms/ios/Ionic Conference App.xcworkspace
+        - scheme: Ionic Conference App
+        - connection: "off"
```

Comment:
Do you know what yaml version are we using or where could this be defined?

Only asking because in YAML 1.1 `on` and `off` are actual bool values while starting from YAML 1.2 only `true` and `false` are bools and `on` and `off` will be left as strings.

Obviously having quotes around `off` is the safest choice so I do not have any problem with it.

Reply:
Oh, I didn't know about this difference in YAML 1.2.

We are using the `yaml.v2` package throughout our packages: https://github.com/bitrise-io/bitrise/blob/master/go.mod#L21

The v3 variant of this package (`yaml.v3`) has this mention [in the readme](https://github.com/go-yaml/yaml/tree/v3) about booleans:

> Specifically, as of v3 of the yaml package:

> YAML 1.1 bools (yes/no, on/off) are supported as long as they are being decoded into a typed bool value. Otherwise they behave as a string. Booleans in YAML 1.2 are true/false only.

Reply:
I would also rather be safe than sorry, but we can revisit this convention in the future and upgrade our YAML parsing as well.